pysph/examples/surface_tension/capillary_wave.py

```diff
@@ -0,0 +1,264 @@
+            dt=dt, tf=tf, adaptive_timestep=False)
+        return solver
+
+    def add_user_options(self, group):
+        choices = ['morris', 'tvf', 'adami_stress', 'adami', 'shadloo']
+        group.add_argument(
+            "--scheme", action="store", dest='scheme', default='morris',
+            choices=choices,
+            help='Specify scheme to use among %s' % choices
+        )
+
+    def create_equations(self):
+        adami_stress_equations = [
+            Group(equations=[
+                SummationDensity(
+                    dest='liquid', sources=[
+                        'liquid', 'wall', 'gas']),
+                SummationDensity(
+                    dest='gas', sources=[
+                        'liquid', 'wall', 'gas']),
+                SummationDensity(
+                    dest='wall', sources=['liquid', 'wall', 'gas'])
+            ]),
+            Group(equations=[
+                TaitEOS(dest='liquid', sources=None, rho0=rho1, c0=c0, gamma=1,
+                        p0=p1),
+                TaitEOS(dest='gas', sources=None,
+                        rho0=rho2, c0=c0, gamma=1, p0=p1),
+                SolidWallPressureBCnoDensity(dest='wall', sources=['liquid',
+                                                                   'gas']),
+            ]),
+            Group(equations=[
+                ColorGradientAdami(dest='liquid', sources=['liquid', 'wall',
+                                                           'gas']),
+                ColorGradientAdami(dest='gas', sources=[
+                              'liquid', 'wall', 'gas']),
+            ]),
+            Group(equations=[ConstructStressMatrix(dest='liquid', sources=None,
+                                                   sigma=sigma, d=2),
+                             ConstructStressMatrix(dest='gas', sources=None,
+                                                   sigma=sigma, d=2)]),
+            Group(
+                equations=[
+                    MomentumEquationPressureGradientAdami(
+                        dest='liquid', sources=['liquid', 'wall', 'gas']),
+                    MomentumEquationPressureGradientAdami(
+                        dest='gas', sources=['liquid', 'wall', 'gas']),
+                    MomentumEquationViscosityAdami(
+                        dest='liquid', sources=['liquid', 'gas']),
+                    MomentumEquationViscosityAdami(
+                        dest='gas', sources=['liquid', 'gas']),
+                    SurfaceForceAdami(
+                        dest='liquid', sources=['liquid', 'wall', 'gas']),
+                    SurfaceForceAdami(
+                        dest='gas', sources=['liquid', 'wall', 'gas']),
+                    SolidWallNoSlipBC(dest='liquid', sources=['wall'], nu=nu1),
+                    SolidWallNoSlipBC(dest='gas', sources=['wall'], nu=nu2),
+                ]),
+                ]
+        return adami_stress_equations
```

Comment:
So this only works for the Adami stress equations?

Reply:
Sir,

Yes, this problem has varying density and various viscosity. Other schemes implemented won't be able to solve this problem.